data/src/main/java/org/apache/iceberg/data/MigrationService.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.data;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadPoolExecutor;
+import org.apache.iceberg.relocated.com.google.common.util.concurrent.MoreExecutors;
+import org.apache.iceberg.relocated.com.google.common.util.concurrent.ThreadFactoryBuilder;
+
+/** Have a separate class for getting ExecutorService to make it testable with static mock */
```

Comment:
Are we sure it is a good idea to add another public class for this? How do we see it being used in the future? Why not use `ThreadPools` instead?

Reply:
When we do static mock, all invocations need to be mocked as well. That includes `ThreadPools` used for `SnapshotProducer` and other places. Having a separate class and mock it is the cleanest way.

Reply:
I am not convinced it justifies adding another public class. What if we overload `listPartition` or whatever method we need in `TableMigrationUtil` to accept `ExecutorService`?

It is an open question whether we should accept `ExecutorService` or `int` in the actions (not procedures), though.

Reply:
@aokolnychyi I finally get time to refactor a bit, accepting `ExecutorService` in the actions, `SparkTableUtil` and  `TableMigrationUtil`.